### PR TITLE
Mh 20  verify account

### PR DESCRIPTION
**Pull request by sharrisoncaktus**
User verification! We add two fields to the UserDetails model: 

* verification_code: A unique code that is emailed to the user and included in the verification link
* verification_completed: A timestamp that, when not null, indicates that user has been verified.

There are two new views to support verification; both views are GET-only and immediately redirect to the dashboard:

* send_account_verification – resets the user's verification code (invalidating any old code) and sends a new verification email.
* verify_account – checks the given verification URL against the User's verification code and responds accordingly.

Verification also adds logic in several places:

* During signup, a verification email is sent to the user
* On the dashboard, if the user is unverified there is an alert message to the user to verify their email address, and a link to re-send the verification email. We also disable the "Upload Your Document" button.
* Before saving the User model, we check to see if the User's email address has changed; if so, we reset the User's verification status. (This can be triggered, for example, by changing the email address while editing the profile)

Because your dev environment probably doesn't send email, the best way to test this is to go through the account creation and editing sequence, and to pull the verification code out of the user model using `python manage.py shell`. Once it's on staging, it should send email, so that can be tested more realistically.

### CODE REVIEW

**Review by nmashton — COMMENTED** [review]
I haven't yet perused the test code to see if it makes sense (it runs, that much I can say for sure!), but this looks pretty darn good.

The one problem I see is something I pushed up a commit to fix: the scheme and domain weren't getting used correctly in the email. You need to supply some extra context to the template for that. See if you like what I did, and if you'd rather do it a different way, please feel free to revert my commit and push up your own version!

File: myhpom/tests/test_dashboard_view.py [hunk 1]
```diff
@@ -62,3 +64,45 @@ def test_user_is_organ_donor(self):
...
+
+        # unverified user
+        user.userdetails.reset_verification()
+        user.userdetails.save()
+        self.assertIsNotNone(user.userdetails.verification_code)
+        self.assertIsNone(user.userdetails.verification_completed)
+        response = self.client.get(self.url)
+        messages = [msg for msg in get_messages(response.wsgi_request) if msg.level == WARNING]
+        self.assertGreater(len(messages), 0)
+        matchdata = re.search(
```
Comment (nmashton) [inline myhpom/tests/test_dashboard_view.py]: I think this way of doing it is fine for this PR's purposes, but in the future we might should consider actually parsing the generated HTML and inspecting the resulting element tree—it's a bit more reliable than string searches.

Comment (sharrisoncaktus) [reply myhpom/tests/test_dashboard_view.py]: That's a good idea.

Comment (sharrisoncaktus) [reply myhpom/tests/test_dashboard_view.py]: I'll go ahead and make that change.

**Review by nmashton — APPROVED** [review]

**Pull request merged**